src/components/Post/ExpandedPost.tsx

```diff
@@ -0,0 +1,142 @@
+import classNames from 'classnames';
+import moment from 'moment/moment';
+import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+
+import { convertMarkdownToDraft, DraftEditor } from '../DraftEditor';
+import { fetchLikersByPost, useMeta, usePost, useZKGroupFromPost } from '../../ducks/posts';
+import { useUser } from '../../ducks/users';
+import { PostMessageSubType } from '../../util/message';
+import { getHandle, getUsername } from '../../util/user';
+import { useThemeContext } from '../ThemeContext';
+
+import Avatar from '../Avatar';
+import Nickname from '../Nickname';
+import PostFooter from './PostFooter';
+import PostLikes from './PostLikes';
+import PostMenu from './PostMenu';
+import URLPreview from '../URLPreview';
+import { Props } from './types';
+
+export default function ExpandedPost(
+  props: Props & {
+    onDeletePost: () => void;
+  }
+): ReactElement {
+  const originalPost = usePost(props.messageId);
+  const referencedPost = usePost(originalPost?.payload.reference);
+  const messageId =
+    originalPost?.subtype === PostMessageSubType.Repost
+      ? originalPost.payload.reference
+      : props.messageId;
+  const post = originalPost?.subtype === PostMessageSubType.Repost ? referencedPost : originalPost;
+  const user = useUser(post?.creator);
+  const history = useHistory();
+  const [parentCreator, parentHash] = post?.payload.reference.split('/') || [];
+  const parentUser = useUser(parentCreator);
+  const meta = useMeta(props.messageId);
+  const zkGroup = useZKGroupFromPost(props.messageId);
+  const theme = useThemeContext();
+  const [likers, setLikers] = useState<string[] | null>(null);
+
+  useEffect(() => {
+    fetchLikersByPost(messageId).then(likers => setLikers(likers));
```

Comment:
instead of calling it here, i think we should call it in `PostLikes` modal, that way we only make the request when users want to see the list of likers.

Reply:
I did some refactoring when implementing the followers/followings modal that does just that I think
https://github.com/zkitter/ui/pull/41/files#r1000588852